common/noise/PerlinNoise.cpp

```diff
@@ -2,6 +2,11 @@
 #include "PerlinNoise.h"
 #include "tbb/tbb.h"
 
+#define wrap_left  ( i==0 ? width-1 : i-1 )
```

Comment:
#define is not appropriate for defining constants, and even worse for macros. 


These need to be redone using safer techniques.